plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -53,18 +52,15 @@ We differ from the paper a few ways, mostly leaving things out:
 Functionality
 ------------
 
-PreInlineUncoditional: we don't do it, since we don't bother using usage information.
+PreInlineUnconditional: we don't do it, since we don't bother using usage information.
 We *could* do it, but it doesn't seem worth it. We also don't need to worry about
 inlining nested let-bindings, since we don't inline any.
 
-CallSiteInline: not worth it.
+CallSiteInline: we do it based on ambient arity, not context. (TODO in PLT-1146)
```

Comment:
This distinction isn't right. We need both the _arity_ of the function to inline and the _context_ of the call site in order to do call site inlining of saturated functions.